session_test.go

```diff
@@ -0,0 +1,271 @@
+package ssh
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"testing"
+
+	gossh "golang.org/x/crypto/ssh"
+)
+
+func (srv *Server) serveOnce(l net.Listener) error {
```

Comment:
Ah, this is a nice hack. I'll hopefully be able to cobble together something a bit more robust when the Shutdown pr goes in.